operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourcePredicates.java

```diff
@@ -0,0 +1,54 @@
+package io.javaoperatorsdk.operator.processing.event.internal;
+
+import io.fabric8.kubernetes.client.CustomResource;
+import io.javaoperatorsdk.operator.api.config.ControllerConfiguration;
+
+public final class CustomResourcePredicates {
+
+  private static final CustomResourcePredicate<CustomResource> GENERATION_AWARE =
```

Comment:
I'm not sure this is equivalent to the old generation processing, is this aware o retries for example?